src/public_parameters/disk_cache.rs

```diff
@@ -1,15 +1,79 @@
+use std::collections::HashMap;
 use std::fs::{create_dir_all, File};
 use std::io::{BufReader, BufWriter, Read};
 use std::marker::PhantomData;
 
 use abomonation::{encode, Abomonation};
 use camino::{Utf8Path, Utf8PathBuf};
+use config::{Config, Environment};
 use nova::traits::Group;
+use once_cell::sync::Lazy;
 
 use crate::coprocessor::Coprocessor;
 use crate::proof::nova::{CurveCycleEquipped, PublicParams, G1, G2};
 use crate::public_parameters::error::Error;
 
+/// Contains the Lurk config settings
+/// The config file is located in `$HOME/.config/lurk.toml` or `<current_dir>/.config/lurk.toml` on WASM.
+/// The config syntax is as follows:
+/// ```toml
+/// public_params = "/path/to/params"
+/// ```
+/// Config settings can be individually overridden by an env var prefixed by `LURK`, e.g.:
+/// ```sh
+/// LURK_PUBLIC_PARAMS="/path/to/params"
+/// ```
+pub(crate) static LURK_CACHE_CONFIG: Lazy<HashMap<String, String>> = Lazy::new(|| {
```

Comment:
I don't think this belongs here. It probably should be named `LURK_CONFIG` and placed in the `cli` folder, because it's truly a config to run the CLI.

Reply:
The goal of this PR is to generalize the config beyond the CLI so that the Lurk library and other applications can use it, with the main use case of having an optionally configurable public param location rather than an explicit function argument. As mentioned in #621, when we/I made the config CLI-specific it became cumbersome and unintuitive to manipulate these settings outside of the CLI (e.g. in https://github.com/lurk-lab/lurk-rs/blob/master/src/coprocessor/circom.rs#L51).

Now, given that this config file may contain CLI-specific settings like `proofs`/`commits` and non-CLI-specific settings like `public_params` and `circom` location, it probably does belong somewhere else like the existing `src/config.rs`, which I am thinking of merging into.